scala-generator/src/main/scala/models/play/files/ModelsBodyParsers.scala

```diff
@@ -0,0 +1,47 @@
+package scala.models.play.files
+
+import io.apibuilder.generator.v0.models.InvocationForm
+
+object ModelsBodyParsers {
+
+	val BadRequest = "_root_.play.api.mvc.Results.BadRequest"
+	val BodyParser = "_root_.play.api.mvc.BodyParser"
+	val ExecutionContext = "_root_.scala.concurrent.ExecutionContext"
+	val JsError = "_root_.play.api.libs.json.JsError"
+	val JsReads = "_root_.play.api.libs.json.Reads"
+	val JsValue = "_root_.play.api.libs.json.JsValue"
+
+	def bodyParser(): String = s"""
+		private def bodyParser[A](parser: ${BodyParser}[${JsValue}])(implicit ec: ${ExecutionContext}, rds: ${JsReads}[A]): ${BodyParser}[A] =
+				parser.validate(_.validate[A].asEither.left.map(e => ${BadRequest}(${JsError}.toJson(e))))
+	"""
+
+	def bodyParser(enum: scala.generator.ScalaEnum): String = bodyParser(enum.name, enum.qualifiedName)
+	def bodyParser(model: scala.generator.ScalaModel): String = bodyParser(model.name, model.qualifiedName)
+	def bodyParser(union: scala.generator.ScalaUnion): String = bodyParser(union.name, union.qualifiedName)
+	def bodyParser(suffix: String, tpe: String): String = s"""
+		def bodyParser${suffix}(parser: ${BodyParser}[${JsValue}])(implicit ec: ${ExecutionContext}, rds: ${JsReads}[${tpe}]): ${BodyParser}[${tpe}] =
+				bodyParser[${tpe}](parser)
+	"""
+
+	def contents(form: InvocationForm): String = {
+		val scalaService = scala.generator.ScalaService(form.service)
```

Comment:
Indentation is weird here

Reply:
actually in the whole file

Reply:
If only we had a library to format everything the same :D 
I think the issue is 1 tab, that is represented by 8 spaces on GitHub, instead of 4 spaces. 

Should I
  - update the file
  - add scalafmt, scalariform, or similar to the project
  - ignore as there are better things to focus on

Reply:
Oh tabs! Would be good to standardize on one way to indent :)

Reply:
I really don't mind.

I love formatters because everyone uses the same without thinking about it. I do understand how adding a formatter on a project of this scale, and with this amount of open PRs, could make rebasing a pain. 

Will do my best to use spaces, sorry if I forget a few files.